lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -126,31 +126,28 @@ def skip?(_path)
 
         def parse(file, request_path)
           processed_source = ProcessedSource.new(file, request_path)
-
           SexpBuilder.new(processed_source.ast, file).build
-        rescue CC::Parser::Client::HTTPError => ex
-          unexpected_parse_error!(ex) unless ex.response_status.to_s.start_with?("4")
-
-          CC.logger.warn("Skipping #{processed_source.path} due to #{ex.class}")
-          CC.logger.warn("Response status: #{ex.response_status}")
-          CC.logger.debug { "Contents:\n#{processed_source.raw_source}" }
-          CC.logger.debug { "Response:\n#{ex.response_body}" }
-          nil
-        rescue CC::Parser::Client::NestingDepthError => ex
-          CC.logger.warn("Skipping #{processed_source.path} due to #{ex.class}")
-          CC.logger.warn(ex.message)
-          CC.logger.debug { "Contents:\n#{processed_source.raw_source}" }
-          nil
         rescue => ex
-          unexpected_parse_error!(ex)
+          handle_exception(processed_source, ex)
         end
 
-        def unexpected_parse_error!(ex)
-          CC.logger.error("Error processing file: #{processed_source.path}")
-          CC.logger.error(ex.message)
+        def handle_exception(processed_source, ex)
```

Comment:
I do recall having a conversation around method naming, and @toddmazierski not loving "handle" for a method that sometimes re-raises. If you find that surprising, consider considering another name.

Reply:
interesting. I also considered something more like:

```
rescue => ex
  log_exception(ex)
  if reraise_exception?(ex)
    raise ex
  end
end

def log_exception(processed_source, ex)
end

def reraise_exception?(ex)
end
```
@toddmazierski any preference?